internal/daemonset/daemonset.go

```diff
@@ -0,0 +1,199 @@
+package daemonset
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/test-network-function/cnf-certification-test/internal/clientsholder"
+	"github.com/test-network-function/cnf-certification-test/pkg/provider"
+	"github.com/test-network-function/cnf-certification-test/pkg/tnf"
+	v1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	imageWithVersion = "quay.io/testnetworkfunction/debug-partner:latest"
+	timeout          = 5 * time.Minute
+	nodeExporter     = "node-exporter"
+	containerName    = "container-00"
+)
+
+//nolint:funlen
+func CreateDaemonSetsTemplate(dsName, namespace, containerName, imageWithVersion string) *v1.DaemonSet {
+	dsAnnotations := make(map[string]string)
+	dsAnnotations["debug.openshift.io/source-container"] = containerName
+	dsAnnotations["openshift.io/scc"] = nodeExporter
+	matchLabels := make(map[string]string)
+	matchLabels["name"] = dsName
```

Comment:
As @edcdavid pointed out, you need the tnf's debug label here:
```
matchLabels["test-network-function.com/app"] = "debug"

```

Reply:
right, I did it.